src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -2,273 +2,374 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
 using System.Collections.Generic;
-using System.Collections.Immutable;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Runtime.CompilerServices;
 using System.Threading;
 using System.Threading.Tasks;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Serialization
+namespace Microsoft.CodeAnalysis.Serialization;
+
+internal sealed class SolutionStateChecksums(
+    Checksum attributesChecksum,
+    ChecksumCollection projectChecksums,
+    ChecksumCollection analyzerReferenceChecksums,
+    Checksum frozenSourceGeneratedDocumentIdentity,
+    Checksum frozenSourceGeneratedDocumentText) : IChecksummedObject
```

Comment:
old system would box these all up into objects (and an array), passing to the base type. we'd then index into that array to get the constituent pieces.  now it's just a strongly typed tree.